src/message.rs

```diff
@@ -307,6 +307,8 @@ pub struct Message {
     pub(crate) ephemeral_timer: EphemeralTimer,
     pub(crate) ephemeral_timestamp: i64,
     pub(crate) text: Option<String>,
+    /// The value of the Subject header. Not set for messages that we sent ourselves.
```

Comment:
i think, in general, we should set the subject also for outgoing messages.

if Alice replies to Bob and Alice then adds a second reply to her own message (not so unusual i'd say), it makes some sense to set the subject accordingly. also, for forwarded messages this is needed as otherwise dc_msg_get_subject() does not work for them.

but as the pr is also an improvement without that bit, we can target that and forwarding also in a subsequent pr, [similar to html that time](https://github.com/deltachat/deltachat-core-rust/pull/2151). this is maybe better wrt review.

Reply:
Yes, unfortunately it's not trivial to save the outgoing subject, as the subject is only computed when the message is actually sent out, somewhere deep inside MimeFactory.

Reply:
But yes, you're right, I didn't think about this. And yes, I'll do a second PR.

Reply:
k, another pr then, that makes sense. quoted_msg_subject maybe shoud be computed outside of mimefactory then, or whatever.